cmd/govalid/main.go

```diff
@@ -2,52 +2,33 @@
 package main
 
 import (
-	"flag"
 	"fmt"
 	"os"
 
-	"github.com/gostaticanalysis/codegen/singlegenerator"
+	"github.com/spf13/cobra"
 
 	govalid_pkg "github.com/sivchari/govalid"
-	"github.com/sivchari/govalid/internal/analyzers/govalid"
-	"github.com/sivchari/govalid/internal/analyzers/markers"
-	"github.com/sivchari/govalid/internal/analyzers/registry"
 )
 
-func main() {
-	// Parse version flag
-	var version bool
-
-	flag.BoolVar(&version, "version", false, "print version information")
-	flag.Parse()
-
-	if version {
-		fmt.Printf("govalid version %s\n", govalid_pkg.Version)
-		os.Exit(0)
-	}
-
-	if err := run(); err != nil {
-		panic(err)
-	}
+var rootCmd = &cobra.Command{
+	Use:     "govalid [flags] [packages]",
+	Short:   "govalid generates type-safe validation code for Go structs",
+	Version: govalid_pkg.Version,
+	Args:    cobra.ArbitraryArgs,
+	Run: func(_ *cobra.Command, _ []string) {
+		if err := runGenerator(); err != nil {
+			fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+			os.Exit(1)
+		}
+	},
 }
 
-// run initializes the analyzers and starts the unit checker.
-func run() error {
-	registry := registry.NewRegistry(
-		registry.AddAnalyzers(markers.Initializer()),
-		registry.AddGenerators(govalid.Initializer()),
-	)
-
-	if err := registry.Init(nil); err != nil {
-		return fmt.Errorf("failed to initialize analyzers: %w", err)
-	}
+func init() {
+	rootCmd.AddCommand(migrateCmd)
+}
 
-	govalid, err := registry.Generator(govalid.Name)
-	if err != nil {
-		return fmt.Errorf("failed to get govalid generator: %w", err)
+func main() {
+	if err := rootCmd.Execute(); err != nil {
+		os.Exit(1)
 	}
```

Comment:
cobra can return error that are not the you you create

So here, you should have this


```suggestion
	if err := rootCmd.Execute(); err != nil {
		fmt.Fprint(os.Stderr, err)
		os.Exit(1)
 	}
```